File: forhksong/A.題目/python/A1-版本轉換/A1-2.py
##====================================
##題目說明：async 和 await 關鍵字 (Python 3.5+ 支援)
##問題：async 和 await 關鍵字在 Python 3.5 之前不被支援。
##
##題目要求：請透過LLM及相關tooling，調整程式，使得可以在Java17 環境中運行。
##
##驗證：
##請確認參賽包workshop_tool腳本已運行，其中java-17-app為已準備好的java 17 SDK runtime container，可執行下列指令獲得編譯結果
##
##>> docker run --rm -v $(pwd):/app java-17-app
##>> 會出現Exception in thread "main" java.lang.reflect.InaccessibleObjectException

##內部參考：在 Java 7 中，反射可以輕鬆訪問私有字段，但在 Java 17 的模塊化系統中，這種行為受到了更多限制，會導致 IllegalAccessException。
##====================================

import asyncio

async def main():
    print("Hello, Async!")

asyncio.run(main())  # 錯誤：舊版本不支援 async/await 語法


#可能的錯誤訊息： SyntaxError: invalid syntax

File: forhksong/A.題目/java/A1-版本轉換/A1-2.java
//========================================================================
//題目說明：以下程式只能在較java 10+版本的SDK運行，請透過自動化工具調整為在較舊的SDK也可運行
//
//提示(題目組內部參考)：var 關鍵字在 Java 10 之前的版本中不被支持。
//
//題目Input： ./java/A1-2.java
//
//題目Output： 生成的檔案需為 ./java/A1-2-answer.java
//
//開發環境：
//請確認參賽包workshop_tool腳本已運行，將產生名為java-A1-2 container，可自動將./java/A1-2-answer.java程式在Java 17環境編譯並執行
//
//指令： docker run --rm -v ./java:/host-java-files java-A1-2 
//
//驗證： 校驗腳本將分別對A1-2.java、A1-2-answer.java 兩個程式輸出結果比對，確認.1)編譯結果正確及.2)程式輸出結果一致。
//========================================================================

public class VarExample {
    public static void main(String[] args) {
        var message = "Hello, Java!";
        System.out.println(message);
    }
}



//預期解法：明確指定變數類型，例如將 var 替換為 String。